backend/substrapp/signals/datasample/pre_save.py

```diff
@@ -1,18 +1,26 @@
+import logging
+import shutil
+
 from os import path, link
 from os.path import normpath
-import shutil
 from django.conf import settings
 
 
+logger = logging.getLogger(__name__)
+
+
 def data_sample_pre_save(sender, instance, **kwargs):
-    directory = path.join(getattr(settings, 'MEDIA_ROOT'), 'datasamples/{0}'.format(instance.pk))
+    destination_path = path.join(getattr(settings, 'MEDIA_ROOT'), 'datasamples/{0}'.format(instance.pk))
+    src_path = normpath(instance.path)
 
     # try to make an hard link to keep a free copy of the data
     # if not possible, keep the real path location
     try:
-        shutil.copytree(normpath(instance.path), directory, copy_function=link)
-    except Exception:
-        shutil.rmtree(directory, ignore_errors=True)
+        shutil.copytree(src_path, destination_path, copy_function=link)
+    except Exception as e:
+        logger.error(f'error happend while copying data from {src_path} to {destination_path} {e}')
```

Comment:
It may be better to use `logger.exception` to log the exception :)

Reply:
done